libraries/AP_DDS/AP_DDS_External_Odom.cpp

```diff
@@ -0,0 +1,60 @@
+
+
+#if AP_DDS_ENABLED && defined(HAL_VISUALODOM_ENABLED)
```

Comment:
To fix this, we can add an `AP_DDS_config.h` that makes the decision. Previously the config was done in AP_DDS_Client.h. Build times go faster with the new method. See AP_GPS_config.h. To use it, you need to include the config file. The CPP file that uses it includes the AP_DDS_Client.h. 

```c++
#if(AP_DDS_ENABLED) && !HAL_VISUALODOM_ENABLED
  #error DDS requires visual odom to be enabled
#endif
```

To represent it as a dependency, see `Tools/scipts/build_options.py`. We need to add `AP_DDS_ENABLED`.
#23280 

Since visual odom is enabled on 2MB board, it should be available for DDS too.

Reply:
How should we handle this? 
```c++
#if AP_DDS_ENABLED && HAL_VISUALODOM_ENABLED

#include "AP_DDS_External_Odom.h"
#include "AP_DDS_Type_Conversions.h"
#include <AP_VisualOdom/AP_VisualOdom.h>

void AP_DDS_External_Odom::handle_external_odom(const tf2_msgs_msg_TFMessage& msg)
{ // todo implement
}
#endif
```
```bash
../../libraries/AP_DDS/AP_DDS_External_Odom.cpp:3:23: error: "HAL_VISUALODOM_ENABLED" is not defined, evaluates to 0 [-Werror=undef]
    3 | #if AP_DDS_ENABLED && HAL_VISUALODOM_ENABLED
      |                       ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: some warnings being treated as errors
```

Reply:
Added the `AP_DDS_config.h` file with the help of Randy and it's fixed.